### DISCUSSION TITLE
Renovate does not create an MR for CVE-2024-26143 at all

### DISCUSSION THREAD

**Discussion by stdedos** [renovatebot/renovate#28072] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, latest

### Please tell us more about your question or problem

RenovateBot is not creating a pull request to address a known vulnerability in Rails 7.1.3, as identified by CVE-2024-26143

I have `gem 'rails', '~> 7.1'` in my Gemfile, and `    rails (7.1.3)` in my lockfile.

As known, rails 7.1.3 is vulnerable (https://www.cvedetails.com/cve/CVE-2024-26143/, apologies, I cannot find the gh/ovs result "quickly enough"). So, I'm expecting that Renovate will SCRAM trying to patch this for me.

I have pulled the latest ndjson file, but I cannot find ANY mention of Renovate taking that under consideration.

1. Why is Renovate not creating a pull request for this CVE?
2. What can I fix / do to troubleshoot this further?

<details>
<summary>Config</summary>

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    ":automergeDisabled",
    ":automergeRequireAllStatusChecks",
    ":combinePatchMinorReleases",
    ":ignoreModulesAndTests",
    ":ignoreUnstable",
    ":maintainLockFilesDisabled",
    ":prConcurrentLimit20",
    ":separateMajorReleases",
    "config:best-practices",
  ],
  "packageRules": [
    {
      "description": "Renovate configuration for backend",
      "matchFileNames": ["backend/**"],
      "matchManagers": ["bundler"],
      "addLabels": ["backend"],
      "commitMessagePrefix": "<>: Renovate - ",
      // Backend never pins versions.
      "rangeStrategy": "replace",
    },
    {
      "description": "Mark major version upgrades",
      "matchUpdateTypes": ["major"],
      "commitMessageSuffix": " [Major]",
    },
    // This should be a roundabout way of saying "but please open vulnerability MRs"; see
    // https://github.com/renovatebot/renovate/discussions/15490#discussioncomment-2702421.
    {
      "description": "Stables require explicit approval",
      "matchBaseBranches": [
        ...
      ],
      "dependencyDashboardApproval": true,
    }
  ],
  "includePaths": [
    "backend/**",
  ],
  "enabledManagers": [
    "bundler",
  ],
  "addLabels": ["renovate"],
  "assigneesFromCodeOwners": true,
  "assigneesSampleSize": 1,
  "reviewersFromCodeOwners": true,
  "reviewersSampleSize": 2,
  "dependencyDashboardLabels": ["renovate"],
  "branchPrefix": "renovate/",
  "ignoreTests": false,
  "minimumReleaseAge": "7d",
  // We do not have `GITHUB_COM_TOKEN` set, so we cannot use `fetchChangeLogs`.
  "fetchChangeLogs": "off",
  // Enable and track Vulnerabilities.
  "osvVulnerabilityAlerts": true,
  "dependencyDashboardOSVVulnerabilitySummary": "all",
  "vulnerabilityAlerts": {
    "rangeStrategy": "bump",
  },
  "baseBranches": [
    "master",
    ...
  ]
}
```

</details>

### Logs (if relevant)

<details><summary>Logs</summary>

```console

$ cat renovate-log.ndjson | grep rails | wc -l
12
$ # manually remove baseBranch != master
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","baseBranch":"master","config":{"bundler":[{"registryUrls":["https://rubygems.org"],"deps":[{"depName":"rails","managerData":{"lineNumber":3},"currentValue":"'~> 7.1'","datasource":"rubygems","lockedVersion":"7.1.3","updates":[],"packageName":"rails","versioning":"ruby","warnings":[],"sourceUrl":"https://github.com/rails/rails","registryUrl":"https://rubygems.org","homepage":"https://rubyonrails.org","changelogUrl":"https://github.com/rails/rails/releases/tag/v7.1.3.2","currentVersion":"7.1.3","fixedVersion":"7.1.3"}],"lockFiles":["backend/Gemfile.lock"],"packageFile":"backend/Gemfile"}]},"msg":"packageFiles with updates","time":"2024-03-08T09:35:51.893Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","branchList":["renovate/master-1","renovate/master-2","renovate/master-3","renovate/master-4","renovate/master-5","renovate/master-6","renovate/master-7","renovate/master-8","renovate/master-9"],"renovateBranches":["renovate/configure","renovate/master-1","renovate/master-2","renovate/master-3","renovate/master-4","renovate/master-5","renovate/master-6","renovate/master-ruby-on-rails-packages","renovate/master-8","renovate/master-9","renovate/master-10","renovate/master-11"],"msg":"Branch lists","time":"2024-03-08T09:37:26.558Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","msg":"remainingBranches=renovate/configure,renovate/master-pin-dependencies,renovate/master-ruby-on-rails-packages","time":"2024-03-08T09:37:26.558Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","msg":"findPr(renovate/master-ruby-on-rails-packages, undefined, open)","time":"2024-03-08T09:37:29.627Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":30,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","branchName":"renovate/master-ruby-on-rails-packages","prNo":14368,"prTitle":"<>: Renovate - Pin dependency rails to v7.1.3 (master)","msg":"Autoclosing PR","time":"2024-03-08T09:37:29.633Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","msg":"Deleted remote branch: renovate/master-ruby-on-rails-packages","time":"2024-03-08T09:37:32.518Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","msg":"No local branch to delete with name: renovate/master-ruby-on-rails-packages","time":"2024-03-08T09:37:32.524Z","v":0}
{"name":"renovate","hostname":"h","pid":18,"level":20,"logContext":"VACimz9mO7EpqaS57sp41","repository":"my/repo","urls":{"https://rubygems.org/api/v1/gems/rails.json (GET,200)":1},"totalRequests":143,"msg":"http statistics","time":"2024-03-08T09:37:33.294Z","v":0}


```

</details>

**Comment by Churro** [comment]
That's because the corresponding GitHub security advisory [GHSA-9822-6m93-xqf4](https://github.com/advisories/GHSA-9822-6m93-xqf4) (= CVE-2024-26143) relates to actionpack and not rails. The import into the osv.dev database used the same package reference ([here](https://osv.dev/vulnerability/GHSA-9822-6m93-xqf4)).

**Reply by stdedos** [reply]
Sadly, the dependencyDashboard does not mention "actionpack" either (ik that dependencyDashboard != ndjson, but "for now" I don't have it).

I think also that "that one" is "one of those packages" that go hand-in-hand with "other packages".
i.e., in our case:

1. `rails` MUST be updated to update `actionpack` (with `--conservative` only? idk exactly 😓)
2. There is nothing mentioning any of the identifiers mentioned

_Eventually_, I'll have another run / do / post a related log investigation. If there's something else obvious, feel free to mention it.

**Reply by rarkins** [reply]
If you have lock file maintenance enabled, you should be able to trigger a PR ahead of schedule at any time to update those indirect dependencies

**Reply by stdedos** [reply]
I'm not sure if "this applies" - but we like to have control over our updates. 

Of course maybe the suggested action "just works". However, if possible, I would like that to be an explicit MR raised "unilaterally" by renovatebot* - regardless of the actual action (lock file maintenance and/or updating the ranges)

**Reply by Churro** [reply]
It's not something that renovate could address if the security advisory it consumes doesn't indicate that rails needs to be updated because of a vulnerability in actionpack.

The same [GHSA-9822-6m93-xqf4](https://github.com/rails/rails/security/advisories/GHSA-9822-6m93-xqf4) advisory also exists in the rails repo itself where rails is flagged as the vulnerable component. If you think this is more appropriate, you could send a correction request to the official GitHub advisory, which may eventually lead to security fix PRs for rails itself.

**Reply by stdedos** [reply]
> The same GHSA-9822-6m93-xqf4 advisory also exists in the rails repo itself where rails is flagged as the vulnerable component. If you think this is more appropriate, you could send a correction request to the official GitHub advisory, which may eventually lead to security fix PRs for rails itself.

I'm probably missing something 😅 

I read your response as "the same CVE _also_ targets `rails`". 

If that's the case then, shouldn't renovate try to open two MRs for the same CVE? 

The `actionpack` probably won't be opened (because of `--conservative`), but the `rails` should definitely have an update (bump ranges and/or lockfile maintenance).

**Reply by Churro** [reply, marked as answer]
The reasoning logic isn't as sophisticated as you suspect 😏 

renovate integrates with Dependabot alerts on GitHub and osv.dev. It matches extracted dependencies against available security advisories that usually reference one or more vulnerable packages. In this case, the official GitHub advisory and the one provided by osv.dev state that the vulnerable item is the `actionpack` gem. This is perfectly accurate and if you use this gem directly, you will see a vulnerability fix PR. 

What the official advisories don't say is that `rails` bundles with `actionpack` and that updating `rails` will also update `actionpack`. This is stated in the second advisory which is only available in the `rails` repo and not used for vulnerability analysis. Hence, you get no PR for `rails`. You'd get one for `actionpack` if the bundler manager in renovate would also extract transitive dependencies from `Gemfile.lock`. However, this is not the case currently (as for most managers), so what you could do is to regularly run a lockfile maintenance task so that the lockfile including `actionpack` is kept updated (regardless of vulnerabilities).

**Reply by stdedos** [reply]
https://github.com/github/advisory-database/pull/4070
https://github.com/advisories/GHSA-9822-6m93-xqf4

Let's see ...

**Reply by stdedos** [reply]
_Sadly_, the project-in-question was discontinued - so there will be no more investigation 😕 

Let's say that the missing package from the "main advisory" was actually critical to it happening
